pkg/authorization/authorizer/personal_subjectaccessreview.go

```diff
@@ -2,35 +2,69 @@ package authorizer
 
 import (
 	"bytes"
-	"fmt"
 	"io/ioutil"
 	"net/http"
+	"strings"
 
 	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apiserver/request"
 	"k8s.io/kubernetes/pkg/runtime"
 
 	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
 )
 
-func IsPersonalAccessReview(a Action) (bool, error) {
-	switch extendedAttributes := a.GetRequestAttributes().(type) {
-	case *http.Request:
-		return isPersonalAccessReviewFromRequest(a, extendedAttributes)
+type personalSARRequestInfoResolver struct {
+	// infoFactory is used to determine info for the request
+	infoFactory RequestInfoFactory
+}
 
-	case *authorizationapi.SubjectAccessReview:
-		return isPersonalAccessReviewFromSAR(extendedAttributes), nil
+func NewPersonalSARRequestInfoResolver(infoFactory RequestInfoFactory) RequestInfoFactory {
+	return &personalSARRequestInfoResolver{
+		infoFactory: infoFactory,
+	}
+}
 
-	case *authorizationapi.LocalSubjectAccessReview:
-		return isPersonalAccessReviewFromLocalSAR(extendedAttributes), nil
+func (a *personalSARRequestInfoResolver) NewRequestInfo(req *http.Request) (*request.RequestInfo, error) {
+	requestInfo, err := a.infoFactory.NewRequestInfo(req)
+	if err != nil {
+		return requestInfo, err
+	}
 
-	default:
-		return false, fmt.Errorf("unexpected request attributes for checking personal access review: %v", extendedAttributes)
+	// only match SAR and LSAR requests for personal review
+	if !requestInfo.IsResourceRequest {
+		return requestInfo, nil
+	}
+	if strings.ToLower(requestInfo.Verb) != "create" {
+		return requestInfo, nil
+	}
+	if len(requestInfo.APIGroup) != 0 {
```

Comment:
do the API groupies know the grouped version won't support the special self SubjectAccessReview behavior?